packages/components/pages/f-contact-preferences/src/components/ContactPreferences.vue

```diff
@@ -8,36 +8,41 @@
             has-outline>
             <form @submit.prevent="onFormSubmit">
                 <div
-                    v-for="{ email, key, sms } in preferences"
+                    v-for="{ key, emailEnabled, emailValue, smsEnabled, smsValue } in preferences"
                     :key="key">
-                    <h2 :class="$style['c-contactPreferences-subtitle']">
+                    <h2
+                        :class="$style['c-contactPreferences-subtitle']">
                         {{ $t(`${key}.subtitle`) }}
                     </h2>
-                    <fieldset :class="$style['c-contactPreferences-fieldset']">
+                    <fieldset
+                        :class="$style['c-contactPreferences-fieldset']">
                         <label>
                             <input
                                 type="checkbox"
-                                :disabled="!email.enabled"
-                                :checked="email.value">
+                                :data-test-id="`contactPreferences-${key}-checkbox`"
+                                :disabled="!emailEnabled"
+                                :checked="emailValue"
+                                @change="editPreferenceValue(key, Object.keys({ emailValue })[0], $event.target.checked)">
```

Comment:
What's the advantage of your method over, say, this?
```suggestion
                                @change="editPreferenceValue(key, 'emailValue', $event.target.checked)">
```

Reply:
So I can keep the entry `emailValue` as a **key** variable so when an engineer reads the code this variable is colour coded in there IDE indicating to them that this is variable plus if an engineer needs to rename then this variable then this entry is included in the rename whereas it would not be if a string.